test/unit/manipulation.js

```diff
@@ -500,6 +500,53 @@ QUnit.test( "html(String) tag-hyphenated elements (Bug #1987)", function( assert
 	assert.equal( j.children().text(), "text", "Tags with multiple hypens behave normally" );
 } );
 
+QUnit.test( "html(String) respect the spec on how to handle special and name-terminating chars for tag names (gh-2005)", function( assert ) {
+
+	assert.expect( 160 );
+
+	var wrapper, child, nodeName,
+		nameTerminatingChars = "\x20-\t-\r-\n-\f".split( "-" ),
```

Comment:
Or rather, those cases should also be tested. It looks like `>` is covered implicitly, so maybe just add `buildChild( method, "<" + nodeName + "/>" )` into the helpers.